presto-main/src/main/java/com/facebook/presto/operator/BigintGroupByHash.java

```diff
@@ -85,6 +92,10 @@ public BigintGroupByHash(int hashChannel, boolean outputRawHash, int expectedSiz
 
         valuesByGroupId = new LongBigArray();
         valuesByGroupId.ensureCapacity(hashCapacity);
+
+        // This interface is used for actively reversing memory (push model) for rehash.
+        // The caller can also report memory for the current class (pull model)
+        this.updateMemory = requireNonNull(updateMemory, "updateMemory is null");
```

Comment:
reversing -> reserving
report memory for the current class -> query memory usage on this object